tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/Inheritance.java

```diff
@@ -0,0 +1,149 @@
+            })
+    public Response getReasons() {
+        Map<String, String> map = new TreeMap<>();
+        for (final LocaleInheritanceInfo.Reason r : LocaleInheritanceInfo.Reason.values()) {
+            map.put(r.name(), r.getDescription());
+        }
+        return Response.ok(map).build();
+    }
+
+    @GET
+    @Path("/locale/{localeId}/{hexId}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(
+            summary = "Explain inheritance of an xpath",
+            description = "Looks up the xpath and explains its inheritance")
+    @APIResponses(
+            value = {
+                @APIResponse(
+                        responseCode = "404",
+                        description = "XPath Hex ID not found",
+                        content =
+                                @Content(
+                                        mediaType = "application/json",
+                                        schema = @Schema(implementation = STError.class))),
+                @APIResponse(
+                        responseCode = "200",
+                        description = "Array of possible alt values",
+                        content =
+                                @Content(
+                                        mediaType = "application/json",
+                                        schema =
+                                                @Schema(
+                                                        implementation =
+                                                                InheritanceResponse.class))),
+            })
+    public Response getInheritanceByHex(
+            @Parameter(required = true, example = "aa", schema = @Schema(type = SchemaType.STRING))
+                    @PathParam("localeId")
+                    String localeId,
+            @Parameter(
+                            required = true,
+                            example = "6154e7673c3829ce",
+                            schema = @Schema(type = SchemaType.STRING))
+                    @PathParam("hexId")
+                    String hexId) {
+        CLDRLocale locale = CLDRLocale.getInstance(localeId);
+        if (locale == null) {
+            return STError.badLocale(localeId);
+        }
+        String xpath = getXPathByHex(hexId);
+        if (xpath == null) {
+            return STError.badPath(hexId);
+        }
+
+        final CLDRFile f = sm.getSTFactory().make(locale, true);
+        final List<LocaleInheritanceInfo> paths = f.getPathsWhereFound(xpath);
+        return Response.ok(new InheritanceResponse(paths)).build();
+    }
+
+    private String getXPathByHex(String hexId) {
+        String xpath = null;
+        try {
+            xpath = sm.xpt.getByStringID(hexId);
+        } catch (RuntimeException e) {
+            /*
+             * Don't report the exception. This happens when it simply wasn't found.
+             * Possibly getByStringID, or some version of it, should not throw an exception.
+             */
```

Comment:
this isn't good karma -- exceptions shouldn't be ignored, especially not generic ones like RuntimeException

better would be to add a new method in XPathTable that simply returns null instead of throwing an exception -- it could wrap getByStringID with a catch (or add a catch within getByStringID itself if that doesn't break compatibility with other callers)

Reply:
This is a copy and paste from elsewhere. But agreed

Reply:
Fixed now.